File: python3/1769/Solution.py
class Solution:
    def minOperations(self, boxes: str) -> List[int]:
        count_boxes = len(boxes)
        answer = [0] * count_boxes

        cost_left = 0
        ball_left = 0
        for left_scan_index in range (0, count_boxes - 1):
            if boxes[left_scan_index] == '1':
                ball_left += 1

            cost_left += ball_left

            answer[left_scan_index + 1] += cost_left

        cost_right = 0
        ball_right = 0
        for right_scan_index in range (count_boxes - 1, 0, -1):
            if boxes[right_scan_index] == '1':
                ball_right += 1

            cost_right += ball_right

            answer[right_scan_index - 1] += cost_right

        return answer

File: java/1769/Solution.java
class Solution {
    public int[] minOperations(String boxes) {
        int boxCount = boxes.length();
        int[] answer = new int[boxCount];

        Arrays.fill(answer, 0);

        int leftCost = 0;
        int leftCount = 0;
        for (int leftScanIndex = 0; leftScanIndex < boxCount - 1; leftScanIndex++) {
            if (boxes.charAt(leftScanIndex) == '1') {
                leftCount += 1;
            }

            leftCost += leftCount;

            answer[leftScanIndex + 1] += leftCost;
        }

        int rightCost = 0;
        int rightCount = 0;
        for (int rightScanIndex = boxCount - 1; rightScanIndex > 0; rightScanIndex--) {
            if (boxes.charAt(rightScanIndex) == '1') {
                rightCount += 1;
            }

            rightCost += rightCount;

            answer[rightScanIndex - 1] += rightCost;
        }

        return answer;
    }
}
